x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_events_query.ts

```diff
@@ -102,17 +126,25 @@ export const buildEventsSearchQuery = ({
   searchAfterSortIds,
   sortOrder,
   timestampOverride,
+  shouldDisableTimestampFallback,
   trackTotalHits,
 }: BuildEventsSearchQuery) => {
   const defaultTimeFields = ['@timestamp'];
   const timestamps =
-    timestampOverride != null ? [timestampOverride, ...defaultTimeFields] : defaultTimeFields;
+    timestampOverride != null
+      ? [timestampOverride, ...(shouldDisableTimestampFallback ? [] : defaultTimeFields)]
+      : defaultTimeFields;
   const docFields = timestamps.map((tstamp) => ({
     field: tstamp,
     format: 'strict_date_optional_time',
   }));
 
-  const rangeFilter = buildTimeRangeFilter({ to, from, timestampOverride });
+  const rangeFilter = buildTimeRangeFilter({
+    to,
+    from,
+    timestampOverride,
+    shouldDisableTimestampFallback,
+  });
 
   const filterWithTime: estypes.QueryDslQueryContainer[] = [filter, rangeFilter];
 
```

Comment:
@marshallmain do we need to skip adding `@timestamp` as a sort option (line 160 below) in case we disable fallback to it?

Reply:
I would leave it in as a sort option, because it's still required for all source documents to have `@timestamp` so it should be safe to sort on the field. It provides some value as a secondary sort to break ties if there are documents with the same `timestampOverride` value. Removing the "fallback" part of the time range filter should be sufficient.

Reply:
On second thought, I think if we disable the fallback then it may make sense to completely ignore `@timestamp` in the rule executors, including for sorting. See other comments about `primaryTimestamp` and `secondaryTimestamp` idea.